modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/NavigationMenuResourceImpl.java

```diff
@@ -136,13 +152,23 @@ public Page<NavigationMenu> getSiteNavigationMenusPage(
 					ActionKeys.UPDATE, "putNavigationMenuBatch",
 					SiteNavigationConstants.RESOURCE_NAME, null)
 			).build(),
-			transform(
-				_siteNavigationMenuService.getSiteNavigationMenus(
-					siteId, pagination.getStartPosition(),
-					pagination.getEndPosition(), null),
-				this::_toNavigationMenu),
-			pagination,
-			_siteNavigationMenuService.getSiteNavigationMenusCount(siteId));
+			booleanQuery -> {
+			},
+			filter, SiteNavigationMenu.class.getName(), search, pagination,
+			queryConfig -> queryConfig.setSelectedFieldNames(
+				Field.ENTRY_CLASS_PK),
+			searchContext -> {
+				if (!Validator.isBlank(search)) {
```

Comment:
If the `search` is not blank then shouldn't we be using this in the search?

Reply:
Yeah, actually this code doesn't make sense, I didn't pay atention to it, I will send the changes!